gestalt/__init__.py

```diff
@@ -159,6 +159,16 @@ def build_config(self) -> None:
         self.__parse_dictionary_keys(self.__conf_sets)
         self.__conf_sets = self.__interpolate_keys(self.__conf_sets)
 
+    def __combine_into(
```

Comment:
That requires more changes to gestalt I believe. We currently load each `json` or `yaml` one by one as per they have been added to gestalt. In a general case, would have been better to use `merge(a, b) -> dict`. But for the way gestalt is set up and keeping minimal changes, I think it is better to use this approach. It is similar to the way it was before, but instead of using an `update` built-in function, it uses the private method.

The overriding test is there to check the functionality of this `__combine_into` function if it works correctly. If that `__combine_into` function has any issues, the output would change from the test, since the merging of two dictionaries didn't behave as expected, along with the other tests should be there to check any incorrect behavior. 

Let me know if we still want to change this function, or write more tests. Happy to do so!

Reply:
Ok I went over there several times and from my findings
- Python dictionary merges are not `deep`
- User libs exist, but don't appear to be widely used 
- nothing in the `collections` lib that really help here 


For the testing I agree with Jospeh. The test now looks good, but it's really testing `build_config`, rather than `__combine_into`. Given this doesn't use the state of the Gestalt class, this might be better as a public function in the package, rather than a private method, making testing just a little simpler. @jdechicchis I would love to hear your thoughts.

Reply:
I think a public function would make sense here. If we go that route though, I would call it `deep_update(a, b: dict) -> dict` and it should do a deep merge and return a new dict without modifying the state of the original dicts IMO. Also, the doc would need to be clear on _which_ dict (a or b) takes precedence if keys clash. e.g., given
```
a = {"foo", 1, "bar", 1}
b = {"foo", 2, "bar", 1}
```
is the result of `deep_update(a,b)`
```
{"foo", 2, "bar", 1}
```
or
```
{"foo", 1, "bar", 1}
```
(I lean towards the former, meaning the second dict takes precedence)